src/bitcoind/mod.rs

```diff
@@ -187,6 +194,60 @@ fn wallet_transaction(bitcoind: &BitcoinD, txid: Txid) -> Option<WalletTransacti
         .ok()
 }
 
+fn cpfp(
+    revaultd: Arc<RwLock<RevaultD>>,
+    bitcoind: Arc<RwLock<BitcoinD>>,
+    txids: Vec<Txid>,
+    feerate: f64,
+) -> Result<(), BitcoindError> {
+    let db_path = revaultd.read().unwrap().db_file();
+    assert!(revaultd.read().unwrap().is_manager());
+
+    let mut cpfp_txs = Vec::with_capacity(txids.len());
+    let mut counter = 0;
+    //
+    // sats/vbyte -> sats/WU
+    let sats_wu = feerate / 4.0;
+    // sats/WU -> msats/WU
+    let msats_wu = (sats_wu * 1000.0) as u64;
+
+    // sats/WU -> sats/kWU
+    let sats_kwu = (sats_wu * 1000.0) as u64;
+
+    for txid in txids.iter() {
+        let spend_tx = db_spend_transaction(&db_path, &txid).expect("Database must be available");
+
+        if let Some(unwrap_spend_tx) = spend_tx {
+            // If the transaction is of type SpendTransaction
+            let psbt = unwrap_spend_tx.psbt;
+            if should_cpfp(&bitcoind.read().unwrap(), &psbt, sats_kwu) {
+                cpfp_txs.push(ToBeCpfped::Spend(psbt));
+                counter += 1;
+            }
+        } else {
+            let unvault_pair =
+                db_vault_by_unvault_txid(&db_path, &txid).expect("Database must be available");
+            let unvault_tx = match unvault_pair {
+                Some((_vault, tx)) => tx,
+                None => return Err(BitcoindError::Custom("Unknown Txid.".to_string())),
+            };
+            // The transaction type is asserted to be UnvaultTransaction
+            let psbt = unvault_tx.psbt.assert_unvault();
+            if should_cpfp(&bitcoind.read().unwrap(), &psbt, sats_kwu) {
+                cpfp_txs.push(ToBeCpfped::Unvault(psbt));
+                counter += 1;
+            }
+        }
+    }
+
+    if counter != 0 {
+        cpfp_package(&revaultd, &bitcoind.read().unwrap(), cpfp_txs, msats_wu)
```

Comment:
This function expects sats/kWU. The value is actually correct but the variable name is confusingly wrong